### PR TITLE
Convert Transform tasks to prefect

### PR DESCRIPTION
**Pull request by dgarros**

### CODE REVIEW

File: backend/infrahub/transformations/tasks.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        name=message.repository_name,
+        service=service,
+        repository_kind=message.repository_kind,
+    )
+
+    rendered_template = await repo.render_jinja2_template(
+        commit=message.commit, location=message.template_location, data={"data": message.data}
+    )
+
+    return rendered_template
```
Comment (ogenstad) [inline backend/infrahub/transformations/tasks.py:51]: I'm not completely sure about the location of this file. I don't have a better suggestion just now but I think we should consider the overall structure of our modules soon and consider moving this at that point.

File: backend/tests/unit/api/test_10_transformation_api.py [hunk 2]
```diff
@@ -1,41 +0,0 @@
...
-        response = client.get(
-            "/api/transform/python/transform01",
-            headers=client_headers,
-        )
-
-    assert response.status_code == 200
-    assert response.json() is not None
-    result = response.json()
-
-    assert result == {"KEY1": "value1", "KEY2": "value2"}
```
Comment (ogenstad) [inline backend/tests/unit/api/test_10_transformation_api.py:41]: I don't think we have another test that's validating this endpoint? Looking at what we had above I don't think it provides a lot of value with the mocked data being returned as we provide it as part of the input. We could add a transform to the internal "car-dealership" repo and test it in the same area as we have the TestProposedChangePipelineRepository test class.

**Review by ogenstad — APPROVED** [review]

File: backend/infrahub/transformations/tasks.py (hunk 1 above)
Comment (dgarros) [reply backend/infrahub/transformations/tasks.py:51]: Agreed

File: backend/tests/unit/api/test_10_transformation_api.py (hunk 2 above)
Comment (dgarros) [reply backend/tests/unit/api/test_10_transformation_api.py:41]: Good catch, I've added a new test for this endpoint

**Pull request merged**